third_party/2and3/itsdangerous.pyi

```diff
@@ -47,107 +39,115 @@ class BadHeader(BadSignature):
 
 class SignatureExpired(BadTimeSignature): ...
 
-def base64_encode(string: _can_become_bytes) -> bytes: ...
-def base64_decode(string: _can_become_bytes) -> bytes: ...
+def base64_encode(string: Union[str, bytes]) -> bytes: ...
+def base64_decode(string: Union[str, bytes]) -> bytes: ...
 def int_to_bytes(num: int) -> bytes: ...
-def bytes_to_int(bytestr: _can_become_bytes) -> bytes: ...
+def bytes_to_int(bytestr: Union[str, bytes]) -> bytes: ...
 
-class SigningAlgorithm:
-    def get_signature(self, key: _bytes_like, value: _bytes_like) -> bytes: ...
-    def verify_signature(self, key: _bytes_like, value: _bytes_like, sig: _can_become_bytes) -> bool: ...
+class SigningAlgorithm(object):
+    def get_signature(self, key: bytes, value: bytes) -> bytes: ...
+    def verify_signature(self, key: bytes, value: bytes, sig: Union[bytes, str]) -> bool: ...
 
 class NoneAlgorithm(SigningAlgorithm):
-    def get_signature(self, key: _bytes_like, value: _bytes_like) -> bytes: ...
+    def get_signature(self, key: bytes, value: bytes) -> bytes: ...
 
 class HMACAlgorithm(SigningAlgorithm):
     default_digest_method = ...  # type: Callable
     digest_method = ...  # type: Callable
     def __init__(self, digest_method: Optional[Callable] = ...) -> None: ...
-    def get_signature(self, key: _bytes_like, value: _bytes_like) -> bytes: ...
+    def get_signature(self, key: bytes, value: bytes) -> bytes: ...
 
-class Signer:
+class Signer(object):
     default_digest_method = ...  # type: Callable
     default_key_derivation = ...  # type: str
-    secret_key = ...  # type: _can_become_bytes
-    sep = ...  # type: _can_become_bytes
-    salt = ...  # type: _can_become_bytes
+    secret_key: Union[str, bytes]
+    sep: Union[str, bytes]
+    salt: Union[str, bytes]
     key_derivation = ...  # type: str
     digest_method = ...  # type: Callable
     algorithm = ...  # type: SigningAlgorithm
-    def __init__(self, secret_key: _can_become_bytes, salt: Optional[_can_become_bytes] = ..., sep: Optional[_can_become_bytes] = ...,
+    def __init__(self, secret_key: Union[str, bytes], salt: Optional[Union[str, bytes]] = ...,
```

Comment:
Again, secret_key can be unicode; this is probably true for all `Union[str, bytes]` instances in argument positions.

Reply:
Likely. I just went with what was in the Python 2 stubs for the merge. I will check the actual implementations.

Reply:
@JelleZijlstra Actually the type fixes I made are quite extensive. I could submit them in a separate PR for better reviewability. These changes in this PR will accept everything that was accepted previously (type-wise). Whatever you prefer.

Reply:
I think it's fine to do it here; I can just review the individual commits (as long as you keep itertools around :) ).